pkg/service/model_build_target_group.go

```diff
@@ -483,3 +475,35 @@ func (t *defaultModelBuildTask) buildTargetGroupBindingNodeSelector(_ context.Co
 		MatchLabels: targetNodeLabels,
 	}, nil
 }
+
+func (t *defaultModelBuildTask) buildHealthCheckNetworkingIngressRules(trafficSource, hcSource []elbv2model.NetworkingPeer, tgPort, hcPort intstr.IntOrString,
+	tgProtocol corev1.Protocol, preserveClientIP, customSoureRanges bool) []elbv2model.NetworkingIngressRule {
+	if tgProtocol != corev1.ProtocolUDP &&
+		(hcPort.String() == healthCheckPortTrafficPort || hcPort.IntValue() == tgPort.IntValue()) {
```

Comment:
Nit:
`hcPort.IntValue() == tgPort.IntValue` this check is hacky.
If somehow in the future, both hcPort and tgPort is a string but not equal, this will evaluate to true.  

ideally we should check hcPort's type.

Reply:
currently health check port is limited to int values. Will revisit this logic when we support named ports as well.